src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -567,16 +567,20 @@ private AggregateCoercionNode GetAggregateCoercionNode(UnaryOpKind unaryOpKind,
                             continue;
                         }
 
-                        fieldCoercions.Add(
-                            fromField.Name,
-                            InjectCoercion(
-                                new ScopeAccessNode(IRContext.NotInSource(FormulaType.Build(fromField.Type)), new ScopeAccessSymbol(scope, scope.AddOrGetIndexForField(fromField.Name))),
-                                context,
-                                fromField.Type,
-                                toFieldType));
+                        var innerCoersion = InjectCoercion(new ScopeAccessNode(IRContext.NotInSource(FormulaType.Build(fromField.Type)), new ScopeAccessSymbol(scope, scope.AddOrGetIndexForField(fromField.Name))), context, fromField.Type, toFieldType);
+
+                        if (innerCoersion != null)
+                        {
+                            fieldCoercions.Add(fromField.Name, innerCoersion);
+                        }
                     }
                 }
 
+                if (!fieldCoercions.Any())
```

Comment:
>    if (!fieldCoercions.Any()) [](http://example.com/codeflow?start=12&length=30)

or should CoercionMatrix.GetCoercionKind have returned none and avoided this altogether?

Reply:
Upper level doesn't go into details of the record and this method is the one determining what we should do